modules/registration/proxy_request.rb

```diff
@@ -10,11 +10,24 @@ def global_register(request)
       send_request(proxy_req)
     end
 
+    # we support two way of sending data - either a JSON or url encoded data
     def host_register(request)
-      proxy_req = request_factory.create_post '/register',
-                                              request.body.read,
-                                              headers(request),
-                                              request_params(request)
+      if request.content_type == 'application/x-www-form-urlencoded'
+        # the request has a different content type, ForemanRequestFactory sets content type to json, unless
+        # specified explicitly
+        # also request.params contain the same data that is in request.body, just parsed to hash, so we need
+        # to remove them
+        proxy_req = request_factory.create_post '/register',
+                                                request.body.read,
+                                                headers(request).merge("Content-Type" => "application/x-www-form-urlencoded"),
+                                                request_params(request).tap { |params| params.delete("host") }
```

Comment:
Is this the only param to remove? If we already pass the body, do we even need to pass these?

Reply:
My understanding is that Sinatra creates this, cleaner would be probably disable Sinatra parsing but I don't know how to do this.

Reply:
Right now, I believe it's the only parameter, there will be more in the future (setup rex, setup insigts), but at this moment it's the only param. Also it's the only param causing problems, because it's nested hash, if it's single level, e.g. `setup_rex: false` it properly escaped and doesn't cause issues.

Considering all pros and cons, I think it's better to turn the logic and keep the url parameter, drop everything else (not just the host).